src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -484,6 +503,249 @@ protected void addRead(final int size) {
      * @param client This is the client the close is being called for.
      */
     public void onClose(Client client);
-  }  
+  }
+  
+  /**
+   * ClientOptions that can be changed depending on what kind of client you want.
+   * 
+   * In general these should not be set unless you have a very specific use case.
+   * 
+   * @author lwahlmeier
+   *
+   */
+  public interface ClientOptions {
+    
+    /**
+     * This is only available for connection backed by a TCP socket.
+     * It will turn on TcpNoDelay on the the connection at the System level.
+     * This means that data queued to go out the socket will not delay before being sent.
+     * 
+     * @param enabled true means NoDelay is on, false means NoDelay is off.
+     * @return true if this was able to be set.
```

Comment:
I originally had enum + value, but I really didnt like how it worked.  The true false is almost exclusively because of the set send/recv buffer sizes.  Different OSes allow different sizes for min/max and there is no way easy way to get them before hand so all I can do is tell you if it worked or not.  I just kept the same thing for the other calls to keep things the same but they pretty much always just set.